include/host-build.mk

```diff
@@ -205,6 +205,19 @@ ifndef DUMP
 endif
 
 define HostBuild
-  $(HostBuild/Core)
-  $(if $(if $(PKG_HOST_ONLY),,$(if $(and $(filter host-%,$(MAKECMDGOALS)),$(PKG_SKIP_DOWNLOAD)),,$(STAMP_PREPARED))),,$(if $(strip $(PKG_SOURCE_URL)),$(call Download,default)))
+  _host_tools_already_installed:=false
+
+  ifeq ($(CONFIG_IN_SDK),y)
+    ifneq ($(wildcard $(HOST_STAMP_INSTALLED)),)
+      _host_tools_already_installed:=true
+      $$(warning $(PKG_NAME) host part is already built, to force re-compilation remove $(HOST_STAMP_INSTALLED))
+      host-compile:
+		# already compiled, do nothing
```

Comment:
Instead of using `_host_tools_already_installed` can't we just print the warn and continue (and skip the HostBuild/Core and other...)?

It seems strange to introduce a variable just for an early return. Also why host-compile is set?

Reply:
Thanks for the review!

> Instead of using _host_tools_already_installed can't we just print the warn and continue (and skip the HostBuild/Core and other...)?
> It seems strange to introduce a variable just for an early return.

I agree with you! Unfortunately, I couldn't find an easy way to do this in GNU Make. As far as I'm aware, GNU Make doesn't have an early return/continue function. And I can't seem to see an easy way to use `$(and` and `$(if` to combine both the `ifeq` checks into a single condition (at least not without using `$(shell` to use bash for it, but then we need to worry about shell injection bugs).

We could do something like the following, but then we're duplicating the `$(HostBuild/Core)` and other stuff:

```make
  ifeq ($(CONFIG_IN_SDK),y)
    ifneq ($(wildcard $(HOST_STAMP_INSTALLED)),)
      $$(warning $(PKG_NAME) host part is already built, to force re-compilation remove $(HOST_STAMP_INSTALLED))
      host-compile:
		# already compiled, do nothing
    else
      $(HostBuild/Core)
      $(if $(if $(PKG_HOST_ONLY),,$(if $(and $(filter host-%,$(MAKECMDGOALS)),$(PKG_SKIP_DOWNLOAD)),,$(STAMP_PREPARED))),,$(if $(strip $(PKG_SOURCE_URL)),$(call Download,default)))
    endif
  else
    $(HostBuild/Core)
    $(if $(if $(PKG_HOST_ONLY),,$(if $(and $(filter host-%,$(MAKECMDGOALS)),$(PKG_SKIP_DOWNLOAD)),,$(STAMP_PREPARED))),,$(if $(strip $(PKG_SOURCE_URL)),$(call Download,default)))
  endif
```

I'm not an expert with GNU Make however, so if there's a better way of doing it, let me know!

> Also why host-compile is set?

Building `ncurses [host]` fails without `host-compile`:

```console
root@8b9962f4ee63:/home/openwrt-sdk-ath79-generic_gcc-11.3.0_musl.Linux-x86_64# make -j1 V=scw
<Output SNIPPED for readability>
make[3]: Entering directory '/home/openwrt-sdk-ath79-generic_gcc-11.3.0_musl.Linux-x86_64/feeds/base/package/libs/ncurses'
Makefile:187: ncurses host part is already built, to force re-compilation remove /home/openwrt-sdk-ath79-generic_gcc-11.3.0_musl.Linux-x86_64/staging_dir/hostpkg/stamp/.ncurses_installed
make[3]: *** No rule to make target 'host-compile'.  Stop.
make[3]: Leaving directory '/home/openwrt-sdk-ath79-generic_gcc-11.3.0_musl.Linux-x86_64/feeds/base/package/libs/ncurses'
time: package/feeds/base/ncurses/host-compile#0.19#0.09#0.25
    ERROR: package/feeds/base/ncurses [host] failed to build.
make[2]: *** [package/Makefile:114: package/feeds/base/ncurses/host/compile] Error 1
```

Reply:
I hope you don't hate me but i'm checking the makefile... Wonder if we should operate on the HOST_STAMP_INSTALLED function directly? 

Can you do some test by adding some if there? By making the changes there in theory we should be able to have (relatively) cleaner code.

Or HostBuild/Core can't be declared or that would cause some problem in a SDK env?

Reply:
Should I rebase the `fixup!` commit into a single commit/patch now? The `Test Formalities` CI action is throwing an error at me.

I can't seem to find anything in https://openwrt.org/docs/guide-developer/working-with-github-pr#squashing_commits about _when_ to squash commits, but from some other open-source repos explicitly say only to rebase/squash commits when a reviewer says to, since otherwise GitHub's PR UI makes things messy.

Reply:
yes pls, squash them... you can ignore the ci failing, will be autofixed with the squash ;D